lib/buffer.js

```diff
@@ -807,8 +807,8 @@ Buffer.prototype.toJSON = function() {
 
 
 function adjustOffset(offset, length) {
-  offset |= 0;
-  if (offset === 0) {
+  offset = Math.trunc(+offset);
```

Comment:
Is the `+` here still necessary? It seems `Math.trunc()` handles conversions already?